core/ide/src/main/java/org/alice/stageide/sceneeditor/StorytellingSceneEditor.java

```diff
@@ -468,20 +470,27 @@ public void setSelectedExpression(Expression expression) {
     }
   }
 
+  private double clampValue(double min, double max, double val) {
+    return Math.min(max, Math.max(min, val));
```

Comment:
It's a style choice, but I almost feel like you want a pictureClamp and an cameraClamp, which each have their own defaults inside, since using this function makes all your lines long and harder to read

Reply:
thx, define two new functions that use this clampValue function